### PR TITLE
Feature/103 vda name registry update on audit

### LINKED ISSUE #103: [vda-name-registry] Update on audit report
## High

- Potential data corruption caused by updating `max names per DID1

## Low

- Misleading naming in findDID function
- Lack of check on maxNamesPerDID

### PR DESCRIPTION
**Pull request by ITStar10**

### CODE REVIEW

File: VDA-Name-Registry/contracts/NameRegistry.sol [hunk 1]
```diff
@@ -177,11 +233,12 @@ contract NameRegistry is  OwnableUpgradeable {
      * Will be rejected if suffix already registered
      * @param suffix - Suffix to be added
      */
-
-    function addSufix(string memory suffix) external onlyOwner {
+    function addSufix(string memory suffix) external payable onlyOwner {
```
Comment (tahpot) [inline VDA-Name-Registry/contracts/NameRegistry.sol]: Typo. Should be `addSufix`

File: VDA-Name-Registry/contracts/VeridaDataVerificationLib.sol [hunk 2]
```diff
@@ -16,8 +16,13 @@ library VeridaDataVerificationLib {
     bytes memory _unsignedData,
     bytes calldata _signature,
     address _signerAddress
-  ) internal pure returns (bool) {
+  ) internal pure returns (bool result) {
```
Comment (tahpot) [inline VDA-Name-Registry/contracts/VeridaDataVerificationLib.sol]: Shouldn't this be deleted and the re-usable library used instead?

**Review by tahpot — CHANGES_REQUESTED** [review]
See feedback.

File: VDA-Name-Registry/contracts/VeridaDataVerificationLib.sol (hunk 2 above)
Comment (ITStar10) [reply VDA-Name-Registry/contracts/VeridaDataVerificationLib.sol]: No, we use separate VeridaDataVerficationLib for `DID-Registry` and `NameRegistry` contracts.

File: VDA-Name-Registry/contracts/NameRegistry.sol (hunk 1 above)
Comment (ITStar10) [reply VDA-Name-Registry/contracts/NameRegistry.sol]: Will update.

**Review by tahpot — APPROVED** [review]

**Pull request merged**